etl/scripts-ccao-data-warehouse-us-east-1/spatial/spatial-ccao-corner.R

```diff
@@ -0,0 +1,236 @@
+library(arrow)
+library(aws.s3)
+library(ccao)
+library(dplyr)
+library(geoarrow)
+library(glue)
+library(here)
+library(osmdata)
+library(purrr)
+library(sf)
+library(tictoc)
+
+# This script detects corner lots in Cook County parcels and saves a boolean
+# indicator as well as the cross used by the corner detection algorithm
+AWS_S3_RAW_BUCKET <- Sys.getenv("AWS_S3_RAW_BUCKET")
+AWS_S3_WAREHOUSE_BUCKET <- Sys.getenv("AWS_S3_WAREHOUSE_BUCKET")
+output_bucket <- file.path(AWS_S3_WAREHOUSE_BUCKET, "spatial", "ccao", "corner")
+
+# Get the parcel file years for which we should make corner lot indicators
+parcel_path <- file.path(AWS_S3_WAREHOUSE_BUCKET, "spatial", "parcel")
+parcel_years <- open_dataset(parcel_path) %>%
+  distinct(year) %>%
+  collect() %>%
+  pull(year)
+
+# Drop years before 2014, since OSM data for roads is spotty prior to that
+parcel_years <- parcel_years[parcel_years >= 2014]
+
+# Iterate over the years and townships, saving the results to a temporary file
+# after each iteration
+for (iter_year in parcel_years) {
+  tictoc::tic(paste("Finished processing corners for:", iter_year))
+
+  # Load the full year's parcel file to iterate though by township
+  parcels <- open_dataset(parcel_path) %>%
+    filter(year == iter_year) %>%
+    geoarrow_collect_sf()
+
+  for (iter_town in ccao::town_dict$township_code) {
+    print(paste("Now processing township:", iter_town))
+
+    remote_file <- file.path(
+      output_bucket,
+      glue("year={iter_year}"),
+      glue("township_code={iter_town}"),
+      "part-0.parquet"
+    )
+
+    town_parcels <- parcels %>%
+      filter(town_code == iter_town)
+
+    # Get a bounding box 3,000 feet larger than the extent of the town parcels.
+    # Used to query OSM for streets and to ensure parcels on the "edge" of the
+    # township have their correct set of neighbors to check in Step 3 below
+    town_bbox <- town_parcels %>%
+      st_set_geometry("geometry_3435") %>%
+      st_bbox() %>%
+      st_as_sfc() %>%
+      st_buffer(3000, endCapStyle = "FLAT", joinStyle = "MITRE") %>%
+      st_transform(4326) %>%
+      st_bbox()
+
+    # Get a buffered copy of the town parcels. This is used in Step 3 below in
+    # order to fill any tiny gaps between abutting parcels
+    town_parcels_buf <- parcels %>%
+      filter(
+        between(lon, town_bbox$xmin, town_bbox$xmax) &
+          between(lat, town_bbox$ymin, town_bbox$ymax)
+      ) %>%
+      st_buffer(dist = units::set_units(2, "m"))
+
+    # Fetch the OSM street network for the township, removing any OSM way types
+    # that are not main roads
+    osm_streets <- opq(bbox = town_bbox) %>%
```

Comment:
**[Question, non-blocking]** I'm guessing you made this choice intentionally, but is it faster to issue a separate Overpass query for every bounding box vs. once for the whole county? Or is the difference negligible?

Reply:
It indeed seems to be slightly faster to query by township than for the whole county.